### DISCUSSION TITLE
How to turn on the screen remotely

### DISCUSSION THREAD

**Discussion by danielpaulus** [danielpaulus/go-ios#88] [Q&A]
@petemyron posted this as an issue:
Awesome library--thanks for making this!

I'm messing with some iPhones remotely and can't quite complete some WebDriverAgent initialization until the screen is on. I played with this a bit and found that I can turn the screen on with go-ios launch com.apple.Preferences (edit: or com.apple.Home, etc.). I hope that helps others. I'd prefer that the behavior be closer to the power button being pressed instead of launching something.

Assume pairing is complete, can you think of a better way to force the screen to come on with this library? It feels like a hack.

**Comment by danielpaulus** [comment]
./go-ios launch com.apple.springboard

Is a bit less hacky, as it basically launches springboard only :-D

**Comment by krishtoautomate** [comment]
@danielpaulus I see you hacked Quicktimeplayer by mocking the binary to serial port. Is it possible to do same by mocking it with keyboard binary, cheating ios device as wired keyboard connected and mocking keyboard commands

**Reply by danielpaulus** [reply]
One thing you can do is fake a bluetooth keyboard. I have some experimental code for that. Or you can tinker with these iOS dongles: https://www.amazon.de/-/en/Adapter-Digital-Charging-Support-Keyboard/dp/B089FG8WQF and make them work as a keyboard.

**Reply by latorante** [reply]
Hey there @danielpaulus , would you be albe to share your experimental code for fake blueetooth keyboard? I'm interested :)

**Reply by danielpaulus** [reply]
Yes it is totally possible, actually I thought I already made it public. Upon checking I realized I kept it private. Let me publish it.

**Reply by danielpaulus** [reply]
https://github.com/danielpaulus/software-bluetooth-keyboard
here you go. If you want to contribute to it, that would be super cool. You can join my Discord to discuss it :-)

**Reply by latorante** [reply]
Hey @danielpaulus l that's awesome, let me check it out! I'm intrigued. Was more for learning purposes, but if I gain enough skills in go-arena, happy to contribute!
